File: backend/src/Shalom.Application/Contracts/ReadingPlanDto.cs
namespace Shalom.Application.Contracts;

/// <summary>The active reading plan with every day and its completion state.</summary>
public record ReadingPlanDto(
    Guid Id,
    string Name,
    DateOnly StartDate,
    bool IsActive,
    int CompletedCount,
    int TotalDays,
    IReadOnlyList<ReadingDayDto> Days);

File: frontend/projects/api/src/lib/models/reading-plan-dto.ts
import { ReadingDayDto } from './reading-day-dto';

/** The active reading plan with every day and its completion state. */
export interface ReadingPlanDto {
  readonly id: string;
  readonly name: string;
  readonly startDate: string;
  readonly isActive: boolean;
  readonly completedCount: number;
  readonly totalDays: number;
  readonly days: readonly ReadingDayDto[];
}
